### PR TITLE
#496 Pipeline pattern

### PR DESCRIPTION
**Pull request by jjjimenez100**
A sample implementation of the pipeline pattern.
Resources:
* [The Pipeline Pattern — for fun and profit](https://medium.com/@aaronweatherall/the-pipeline-pattern-for-fun-and-profit-9b5f43a98130)
* [The Pipeline design pattern (in Java)](https://medium.com/@deepakbapat/the-pipeline-design-pattern-in-java-831d9ce2fe21)
* [Pipelines | Microsoft Docs](https://docs.microsoft.com/en-us/previous-versions/msp-n-p/ff963548(v=pandp.10))

### CODE REVIEW

**Comment by iluwatar** [comment]
@jjjimenez100 the build is failing due to PMD error. See https://travis-ci.org/iluwatar/java-design-patterns/builds/594261640

**Comment by jjjimenez100** [comment]
Hi @iluwatar. Sorry about that, this is actually my first PR with a repo having checkstyle and PMD.
I tried running `mvn pmd:pmd pmd:cpd` on the pipeline directory I did to try and correct errors it reported. Hopefully this will have a successful build now :ok_hand:

File: pipeline/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+permalink: /patterns/pipeline/
+categories: Behavioral
+tags:
+ - Java
+ - Functional
+ - Difficulty-Intermediate
+---
+
+## Also known as
+None
```
Comment (iluwatar) [inline pipeline/README.md]: We can leave this chapter if there are no aliases

File: pipeline/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+## Intent
+Allows processing of data in a series of stages by giving in an initial input and passing the processed output to be used by the next stages.
+
+## Applicability
+Use the Pipeline pattern when you want to
+
+* execute individual stages that yields a final value
+* add readability to complex sequence of operations by providing a fluent builder as an interface
+* improve testability of code since stages will most likely be doing a single thing, complying to the Single Responsibility Principle (SRP)
```
Comment (iluwatar) [inline pipeline/README.md]: We can link to https://java-design-patterns.com/principles/#single-responsibility-principle

**Comment by iluwatar** [comment]
Thanks @jjjimenez100, beautiful pattern and well implemented :+1:

**Comment by oconnelc** [comment]
I don't know if this is related, but the pipeline project is failing to compile for me because of the project hierarchy. 

It's currently:

src/main/java/com.iluwatar.pipeline/*.java

I think it should be:
src/main/java/com/iluwatar/pipeline/*.java

**Comment by oconnelc** [comment]
Also I have a commit that I can push up as a PR.

**Comment by iluwatar** [comment]
@oconnelc please create an issue about the problem, and then the PR that fixes this

**Pull request merged**